osu.Game.Rulesets.Osu/Mods/OsuModSpacingAdjust.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Bindables;
+using osu.Game.Beatmaps;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Beatmaps;
+using osu.Game.Rulesets.Osu.Utils;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    public class OsuModSpacingAdjust : Mod, IApplicableToBeatmap
+    {
+        public override string Name => "Spacing Adjust";
+
+        public override string Description => "Adjust object spacing to your liking.";
+
+        public override double ScoreMultiplier => 1;
+
+        public override string Acronym => "SA";
+
+        public override ModType Type => ModType.Conversion;
+
+        public override bool RequiresConfiguration => true;
+
+        [SettingSource("Object spacing", "Modifies the spacing between objects.")]
+        public BindableNumber<float> ObjectSpacing { get; } = new BindableFloat
+        {
+            MinValue = 0.5f,
+            MaxValue = 2,
+            Default = 1,
+            Value = 1,
+            Precision = 0.01f,
+        };
+
+        public override string SettingDescription => ObjectSpacing.IsDefault ? string.Empty : $"{ObjectSpacing.Value:N2}x";
+
+        public void ApplyToBeatmap(IBeatmap beatmap)
+        {
+            if (!(beatmap is OsuBeatmap osuBeatmap))
+                return;
+
+            var positionInfos = OsuHitObjectGenerationUtils.GeneratePositionInfos(osuBeatmap.HitObjects);
+
+            positionInfos.ForEach(p => p.DistanceFromPrevious *= ObjectSpacing.Value);
```

Comment:
I think you may want to exclude some objects here for better results. For instance, the very first object on a map has a `DistanceFromPrevious` relative to the playfield centre. So applying object spacing onto it makes it such that the first object will actually be farther from the centre than on the original map, which seems unintended because there is no "object spacing" to speak of yet, really, on the first object.

SImilar arguments could be made for objects immediately succeeding spinners, but that case may be a lot trickier to handle than just skipping the spinner (you'd ideally want to keep spinners' absolute position the same as it was, which is going to be a pain given that this generation process is based entirely off of relative distances and angles).